include/zephyr/math/util.h

```diff
@@ -0,0 +1,104 @@
+/**
+ * @file math/util.h
+ *
+ * @brief A common math library for use in both floating and fixed point.
+ */
+
+/*
+ * Copyright (c) 2022 Google Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef INCLUDE_MATH_UTIL_H_
+#define INCLUDE_MATH_UTIL_H_
+
+#include <stdlib.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifdef CONFIG_FPU
+#define PRIf	    "f"
+#define PRIf_ARG(n) (n)
+#else
+#define PRIf		    "d.%06llu"
+#define __FP_RAW_DECIMAL(n) ((int16_t)((n) & ((1 << FP_BITS) - 1)))
+#define __FP_DECIMAL(n)	    ((n < 0) ? (~__FP_RAW_DECIMAL(n) + 1) : __FP_RAW_DECIMAL(n))
+#define PRIf_ARG(n)                                                                                \
+	(FP_TO_INT(abs(n)) * (n < 0 ? -1 : 1)),                                                    \
+		((__FP_DECIMAL(n) * UINT64_C(1000000)) + ((1 << FP_BITS) / 2)) / (1 << FP_BITS)
+#endif
+
+#ifdef CONFIG_FPU
```

Comment:
I do not like the fact that this library silently uses floats when the FPU is present. What about compatibility of the data between different cores? What if one core has FPU and the other does not and they are exchanging the data? Fixed point should always be fixed point and the decision if we should use floats or not should be up to the module that uses it.